frontend/app/controllers/Event.scala

```diff
@@ -134,13 +134,12 @@ class Event(
   }
 
   def buy(id: String): Action[AnyContent] = eventbriteService.getEvent(id) match {
-      case Some(event@(_: GuLiveEvent)) =>
-        BuyAction(id).async { implicit request =>
-          if (event.isBookableByTier(request.subscriber.subscription.plan.tier))
-            redirectSignedInMemberToEventbrite(event)
-          else suggestUserUpgrades
-        }
-      case Some(event@(_: MasterclassEvent)) =>
+      // One conditional now covers Live Events AND Masterclasses...
+
+      // Logged out readers and non-members will go via the onUnauthenticated route
+      // They, and any ineligible-tier Members will get a "SOLD OUT" EventBrite page if the event has membership-only tickets.
+      // Previously ineligible-tier Members would get a suggested upgrade page.
+      case Some(event@(_: RichEvent)) =>
         BuyAction(id, onUnauthenticated = redirectAnonUserToEventbrite(event)(_)).async { implicit request =>
           redirectSignedInMemberToEventbrite(event)
```

Comment:
Are _redirectAnonUserToEventbrite_ and _redirectAnonUserToEventbrite_ still sensible names for these functions?

Reply:
Given your comment it feels like these should be _redirectEligibleSignedInUserToEventbrite_ and _redirectIneligibleUserToEventbrite_ (or something along those lines)?

Reply:
Sorry, I misread this, feel free to ignore these comments.

Reply:
How wbout redirectMemberToEventBrite and redirectNonMemberToEventBrite ?

Reply:
Yes, perhaps that's a bit clearer 👍